interview-150/274.ts

```diff
@@ -0,0 +1,11 @@
+function hIndex(citations: number[]): number {
+    const sortedArr = citations.sort((a, b) => b - a)
```

Comment:
Note: The `sort` method mutates the original array in place & returns a reference to it. Thus, `sortedArra` here is actually a reference to the same array as `citations`, and the array passed in is sorted as a result of running this function.

That could be desirable or not matter - but isn't clear by looking at the code here. Generally unless performance is cruicial, it is considered best practice to write[ "pure functions"](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976).

To do that, you'd do something like this:
```ts
     const sortedArr = [...citations].sort((a, b) => b - a)
```

Alternatively if performance is crucial, you can make it more obvious that the sort is being done in-place:
```ts
citations.sort((a, b) => b - a)
```